src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +39,396 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+    Vec3f acceleration;
+    Vec3f velocity;
+    Vec3f position;
+    f32 cosValue;
+    s32 i = 0;
+    f32 randomValue;
+    static s16 D_808B85D0 = 0;
+    f32 sinValue;
+    s32 arraySize = 2;
+
+    while (i != arraySize) {
+        if (globalCtx) {}
+        if (!(arg2 < Math_Rand_ZeroOne())) {
+            D_808B85D0 += 30000;
+
+            sinValue = Math_Sins(D_808B85D0);
+            cosValue = Math_Coss(D_808B85D0);
+
+            randomValue = (Math_Rand_ZeroOne() * 35.0f) + 35.0f;
+
+            position.x = (randomValue * sinValue) + this->dyna.actor.posRot.pos.x;
+            position.y = this->dyna.actor.posRot.pos.y + 10.0f;
+            position.z = (randomValue * cosValue) + this->dyna.actor.posRot.pos.z;
+
+            velocity.x = sinValue;
+            velocity.y = 0.0f;
+            velocity.z = cosValue;
+
+            acceleration.x = 0.0f;
+            acceleration.y = 0.5f;
+            acceleration.z = 0.0f;
+
+            func_800286CC(globalCtx, &position, &velocity, &acceleration, ((Math_Rand_ZeroOne() * 16) + 80),
+                          ((Math_Rand_ZeroOne() * 30) + 80));
+        }
+        i++;
+    }
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static s16 D_808B85E4[] = { 61536, 800, 4000 };
+
+void BgSpot18Basket_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+    Actor* actor = &this->dyna.actor;
+    s32 moveFlag = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, 3);
+    func_808B7710(this, globalCtx);
+    DynaPolyInfo_Alloc(&D_06002154, &moveFlag);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, actor, moveFlag);
+
+    Actor_ProcessInitChain(actor, sInitChain);
+    ActorShape_Init(&actor->shape, 0.0f, ActorShadow_DrawFunc_Circle, 15.0f);
+    actor->initPosRot.pos.y = actor->initPosRot.pos.y + 0.01f;
+    actor->posRot.pos.y = actor->initPosRot.pos.y;
+
+    if (Flags_GetSwitch(globalCtx, (actor->params >> 8) & 0x3F)) {
+        func_808B7BB0(this);
+        return;
+    }
+
+    func_808B7AEC(this);
+    Actor_SpawnAsChild(&globalCtx->actorCtx, actor, globalCtx, 451, actor->posRot.pos.x, actor->posRot.pos.y,
+                       actor->posRot.pos.z, actor->shape.rot.x, actor->shape.rot.y + 5461, actor->shape.rot.z, -1);
+
+    if (!actor->child) {
+        osSyncPrintf("\x1b[31m");
+        osSyncPrintf("Ｅｒｒｏｒ : 変化壷蓋発生失敗(%s %d)\n", "../z_bg_spot18_basket.c", 351);
+        osSyncPrintf("\x1b[m");
+        Actor_Kill(actor);
+    }
+}
+
+void BgSpot18Basket_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
```

Comment:
```suggestion
    BgSpot18Basket* this = THIS;
    
    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
```
empty line after declarations